File: Views/Onboarding/CommunicationEvaluationStepView.swift
import SwiftUI

struct CommunicationEvaluationStepView: View {
    @ObservedObject var viewModel: OnboardingViewModel
    
    private let communicationOptions = [
        "1-4",
        "4-6",
        "6-8",
        "8-10"
    ]
    
    var body: some View {
        VStack(spacing: 0) {
            // Espace entre barre de progression et titre (harmonisé)
            Spacer()
                .frame(height: 40)
            
            // Titre centré à gauche
            HStack {
                Text("communication_evaluation_question".localizedOnboarding)
                    .font(.system(size: 28, weight: .bold))
                    .foregroundColor(.black)
                    .multilineTextAlignment(.leading)
                Spacer()
            }
            .padding(.horizontal, 30)
            
            // Sous-titre explicatif
            HStack {
                Text("private_answer_note".localizedOnboarding)
                    .font(.system(size: 14, weight: .regular))
                    .foregroundColor(.black.opacity(0.6))
                    .multilineTextAlignment(.leading)
                Spacer()
            }
            .padding(.horizontal, 30)
            .padding(.top, 8)
            
            Spacer()
            
            // Options de réponse
            VStack(spacing: 12) {
                ForEach(Array(communicationOptions.enumerated()), id: \.offset) { index, option in
                    Button(action: {
                        print("🔥 CommunicationEvaluationStepView: Option sélectionnée: \(option)")
                        viewModel.communicationRating = option
                    }) {
                        Text(option)
                            .font(.system(size: 16))
                            .foregroundColor(viewModel.communicationRating == option ? .white : .black)
                            .multilineTextAlignment(.leading)
                            .lineLimit(nil)
                            .fixedSize(horizontal: false, vertical: true)
                            .frame(maxWidth: .infinity, alignment: .leading)
                            .padding(.vertical, 16)
                            .padding(.horizontal, 20)
                            .background(
                                RoundedRectangle(cornerRadius: 12)
                                    .fill(viewModel.communicationRating == option ? Color(hex: "#FD267A") : Color.white)
                                    .overlay(
                                        RoundedRectangle(cornerRadius: 12)
                                            .stroke(
                                                viewModel.communicationRating == option ? Color(hex: "#FD267A") : Color.black.opacity(0.1),
                                                lineWidth: viewModel.communicationRating == option ? 2 : 1
                                            )
                                    )
                                    .shadow(color: Color.black.opacity(0.05), radius: 8, x: 0, y: 4)
                            )
                    }
                    .buttonStyle(PlainButtonStyle())
                }
            }
            .padding(.horizontal, 30)
            
            Spacer()
            
            // Zone blanche collée en bas avec bouton "Continuer"
            VStack(spacing: 0) {
                Button(action: {
                    print("🔥 CommunicationEvaluationStepView: Bouton continuer pressé")
                    viewModel.nextStep()
                }) {
                    Text("continue".localized)
                        .font(.system(size: 18, weight: .semibold))
                        .foregroundColor(.white)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 16)
                        .background(Color(hex: "#FD267A"))
                        .cornerRadius(28)
                        .opacity(viewModel.communicationRating.isEmpty ? 0.5 : 1.0)
                }
                .disabled(viewModel.communicationRating.isEmpty)
                .padding(.horizontal, 30)
            }
            .padding(.vertical, 30)
            .background(Color.white)
        }
        .background(Color(red: 0.97, green: 0.97, blue: 0.98))
    }
}

#Preview {
    CommunicationEvaluationStepView(viewModel: OnboardingViewModel())
}

File: Views/Onboarding/CommunicationEvaluationStepView.kt
// CommunicationEvaluationStepScreen.kt
package com.love2love.onboarding.ui

import android.util.Log
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.love2love.R

private val Pink = Color(0xFFFD267A)
private val ScreenBg = Color(0xFFF7F7FA)

/**
 * Adaptateur ViewModel -> UI.
 * Hypothèse : viewModel.communicationRating est un état Compose (mutableStateOf)
 * et nextStep() avance l’onboarding.
 *
 * Si ton VM expose un StateFlow ou LiveData, adapte en utilisant collectAsState()/observeAsState().
 */
@Composable
fun CommunicationEvaluationStepScreen(
    viewModel: OnboardingViewModel
) {
    CommunicationEvaluationStepContent(
        selected = viewModel.communicationRating,
        onSelect = { option ->
            Log.d("CommunicationStep", "Option selected: $option")
            viewModel.communicationRating = option
        },
        onContinue = {
            Log.d("CommunicationStep", "Continue pressed")
            viewModel.nextStep()
        }
    )
}

/**
 * Composable stateless : plus simple à tester et à prévisualiser.
 * Tu peux l’utiliser même sans ViewModel (injection d’état par paramètres).
 */
@Composable
private fun CommunicationEvaluationStepContent(
    selected: String,
    onSelect: (String) -> Unit,
    onContinue: () -> Unit
) {
    val options = remember { listOf("1-4", "4-6", "6-8", "8-10") }

    Scaffold(
        containerColor = ScreenBg,
        bottomBar = {
            // Zone blanche collée en bas avec bouton "Continuer"
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .background(Color.White)
                    .padding(vertical = 30.dp)
            ) {
                val enabled = selected.isNotEmpty()
                Button(
                    onClick = onContinue,
                    enabled = enabled,
                    shape = RoundedCornerShape(28.dp),
                    colors = ButtonDefaults.buttonColors(
                        containerColor = Pink,
                        disabledContainerColor = Pink.copy(alpha = 0.5f),
                        contentColor = Color.White,
                        disabledContentColor = Color.White
                    ),
                    modifier = Modifier
                        .padding(horizontal = 30.dp)
                        .fillMaxWidth()
                        .heightIn(min = 48.dp)
                ) {
                    Text(
                        text = stringResource(id = R.string.action_continue),
                        fontSize = 18.sp,
                        fontWeight = FontWeight.SemiBold
                    )
                }
            }
        }
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .padding(innerPadding)
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
        ) {
            Spacer(Modifier.height(40.dp))

            // Titre
            Text(
                text = stringResource(id = R.string.communication_evaluation_question),
                fontSize = 28.sp,
                fontWeight = FontWeight.Bold,
                color = Color.Black,
                modifier = Modifier.padding(horizontal = 30.dp)
            )

            // Sous-titre
            Text(
                text = stringResource(id = R.string.private_answer_note),
                fontSize = 14.sp,
                fontWeight = FontWeight.Normal,
                color = Color.Black.copy(alpha = 0.6f),
                modifier = Modifier
                    .padding(horizontal = 30.dp)
                    .padding(top = 8.dp)
            )

            Spacer(Modifier.height(16.dp))

            // Options
            Column(
                verticalArrangement = Arrangement.spacedBy(12.dp),
                modifier = Modifier.padding(horizontal = 30.dp)
            ) {
                options.forEach { option ->
                    val isSelected = selected == option
                    val shape = RoundedCornerShape(12.dp)

                    Surface(
                        shape = shape,
                        color = if (isSelected) Pink else Color.White,
                        tonalElevation = 0.dp,
                        shadowElevation = 8.dp,
                        modifier = Modifier
                            .fillMaxWidth()
                            .border(
                                width = if (isSelected) 2.dp else 1.dp,
                                color = if (isSelected) Pink else Color.Black.copy(alpha = 0.1f),
                                shape = shape
                            )
                            .clickable { onSelect(option) }
                    ) {
                        Text(
                            text = option,
                            fontSize = 16.sp,
                            color = if (isSelected) Color.White else Color.Black,
                            textAlign = TextAlign.Start,
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(horizontal = 20.dp, vertical = 16.dp)
                        )
                    }
                }
            }

            Spacer(Modifier.height(24.dp))
        }
    }
}

/* --------- Preview sans ViewModel (facultatif) --------- */

@Composable
@androidx.compose.ui.tooling.preview.Preview(showBackground = true)
private fun CommunicationEvaluationStepPreview() {
    var sel by remember { mutableStateOf("") }
    CommunicationEvaluationStepContent(
        selected = sel,
        onSelect = { sel = it },
        onContinue = {}
    )
}

/* --------- Contrat minimal du ViewModel attendu ---------
   Ajuste selon ton implémentation réelle (StateFlow, LiveData, etc.)
*/
interface OnboardingViewModel {
    var communicationRating: String
    fun nextStep()
}
